python_modules/dagster/dagster_tests/scheduler_tests/test_cron_string_iterator.py

```diff
@@ -479,23 +483,91 @@ def test_cron_iterator_leap_day():
+            create_pendulum_time(2023, 3, 26, 3, 15, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 3, 26, 3, 30, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 3, 26, 3, 45, 0, tz="Europe/Berlin"),
+        ],
+    ),
+    (
+        "Europe/Berlin",
+        "*/15 * * * *",
+        [
+            create_pendulum_time(2023, 10, 29, 1, 0, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 1, 15, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 1, 30, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 1, 45, 0, tz="Europe/Berlin"),
+            create_pendulum_time(
+                2023, 10, 29, 2, 0, 0, tz="Europe/Berlin", dst_rule=pendulum.PRE_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 15, 0, tz="Europe/Berlin", dst_rule=pendulum.PRE_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 30, 0, tz="Europe/Berlin", dst_rule=pendulum.PRE_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 45, 0, tz="Europe/Berlin", dst_rule=pendulum.PRE_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 0, 0, tz="Europe/Berlin", dst_rule=pendulum.POST_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 15, 0, tz="Europe/Berlin", dst_rule=pendulum.POST_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 30, 0, tz="Europe/Berlin", dst_rule=pendulum.POST_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 45, 0, tz="Europe/Berlin", dst_rule=pendulum.POST_TRANSITION
+            ),
+            create_pendulum_time(2023, 10, 29, 3, 0, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 3, 15, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 3, 30, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 3, 45, 0, tz="Europe/Berlin"),
+        ],
+    ),
 ]
 
 
 @pytest.mark.parametrize("execution_timezone,cron_string,times", DST_PARAMS)
-def test_dst_transition_advances(execution_timezone, cron_string, times):
+@pytest.mark.parametrize(
+    "force_croniter",
+    [False, True],
+)
+def test_dst_transition_advances(execution_timezone, cron_string, times, force_croniter):
     # Starting 1 second after each time produces the next tick
+
     for i in range(len(times) - 1):
         orig_start_timestamp = to_timezone(times[i], "UTC").timestamp()
         # first start from the timestamp that's exactly on the interval -
         # verify that it first returns the passed in timestamp, then advances
-        fresh_cron_iter = cron_string_iterator(
-            orig_start_timestamp, cron_string, execution_timezone
-        )
+
+        if force_croniter:
```

Comment:
leave a comment as to motivation behind this (to help future refactor-er know whats up)